src/app/features/swap-form/thunks.ts

```diff
@@ -284,21 +303,38 @@ export const confirmSwapThunk = createAsyncThunk<
   void,
   { state: RootState }
 >('swapForm/confirmSwap', async (_, thunkAPI) => {
-  try {
-    const swapFormState = thunkAPI.getState().swapForm;
-    const amm = swapFormState.amm;
-    if (!amm) {
-      return;
-    }
+  const swapFormState = thunkAPI.getState().swapForm;
+  const amm = swapFormState.amm;
+  if (!amm || !amm.signer) {
+    return;
+  }
+
+  const account = await amm.signer.getAddress();
+  const eventParams = {
+    account,
+    notional: getProspectiveSwapNotional(swapFormState),
+    margin: getProspectiveSwapMargin(swapFormState),
+    isEdit: hasExistingPosition(swapFormState),
+    pool: getAmmProtocol(amm),
+    isFT: getProspectiveSwapMode(swapFormState) === 'fixed',
+  };
 
-    return await amm.swap({
+  try {
+    pushSwapTransactionSubmittedEvent(eventParams);
+    const result = await amm.swap({
       isFT: getProspectiveSwapMode(swapFormState) === 'fixed',
       notional: getProspectiveSwapNotional(swapFormState),
       margin: getProspectiveSwapMargin(swapFormState),
       fixedLow: 1,
       fixedHigh: 999,
     });
+    pushSwapTransactionSuccessEvent(eventParams);
+    return result;
   } catch (err) {
+    pushSwapTransactionFailedEvent({
+      ...eventParams,
+      errorMessage: extractError(err),
```

Comment:
is this the message we show to users? asking because in GA we need a short error message not something like sentry

Reply:
It is short message